scheduler/pkg/kafka/gateway/consumer.go

```diff
@@ -169,6 +174,22 @@ func collectHeaders(kheaders []kafka.Header) map[string]string {
 	return headers
 }
 
+func (kc *InferKafkaConsumer) Produce(msg *kafka.Message, deliveryChan chan kafka.Event) error {
```

Comment:
I understand that this is technically not part of this PR but it feels odd to have a producer in a InferKafkaConsumer struct. Not sure if there is a refactor that could reduce the confusion.

Reply:
will refactor